src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -785,87 +653,17 @@ private static boolean isDirectoryExcluded(String path, List<Pattern> patternsTo
-        options.addOption(OPTION_O_NAME, true, "Sets the output file. Defaults to stdout");
-        options.addOption(OPTION_P_NAME, true, "Loads the properties file");
-        options.addOption(OPTION_S_NAME, true,
-                "Print xpath suppressions at the file's line and column position. "
-                        + "Argument is the line and column number (separated by a : ) in the file "
-                        + "that the suppression should be generated for");
-        options.addOption(OPTION_TAB_WIDTH_NAME, true,
-                String.format("Sets the length of the tab character. Used only with \"-s\" option. "
-                        + "Default value is %s",
-                        DEFAULT_TAB_WIDTH));
-        options.addOption(OPTION_GXS_NAME, OPTION_GENERATE_XPATH_SUPPRESSION_NAME, false,
-                "Generates to output a suppression.xml to use to suppress all violations"
-                        + " from user's config");
-        options.addOption(OPTION_F_NAME, true, String.format(
-                "Sets the output format. (%s|%s). Defaults to %s",
-                PLAIN_FORMAT_NAME, XML_FORMAT_NAME, PLAIN_FORMAT_NAME));
-        options.addOption(OPTION_V_NAME, false, "Print product version and exit");
-        options.addOption(OPTION_T_NAME, OPTION_TREE_NAME, false,
-                "Print Abstract Syntax Tree(AST) of the file");
-        options.addOption(OPTION_CAPITAL_T_NAME, OPTION_TREE_COMMENT_NAME, false,
-                "Print Abstract Syntax Tree(AST) of the file including comments");
-        options.addOption(OPTION_J_NAME, OPTION_JAVADOC_TREE_NAME, false,
-                "Print Parse tree of the Javadoc comment");
-        options.addOption(OPTION_CAPITAL_J_NAME, OPTION_TREE_JAVADOC_NAME, false,
-                "Print full Abstract Syntax Tree of the file");
-        options.addOption(OPTION_D_NAME, OPTION_DEBUG_NAME, false,
-                "Print all debug logging of CheckStyle utility");
-        options.addOption(OPTION_E_NAME, OPTION_EXCLUDE_NAME, true,
-                "Directory path to exclude from CheckStyle");
-        options.addOption(OPTION_X_NAME, OPTION_EXCLUDE_REGEXP_NAME, true,
-                "Regular expression of directory to exclude from CheckStyle");
-        options.addOption(OPTION_EXECUTE_IGNORED_MODULES_NAME, false,
-                "Allows ignored modules to be run.");
-        options.addOption(OPTION_CAPITAL_C_NAME, OPTION_CHECKER_THREADS_NUMBER_NAME, true,
-                "(experimental) The number of Checker threads (must be greater than zero)");
-        options.addOption(OPTION_CAPITAL_W_NAME, OPTION_TREE_WALKER_THREADS_NUMBER_NAME, true,
-                "(experimental) The number of TreeWalker threads (must be greater than zero)");
-        return options;
-    }
-
-    /** Helper structure to clear show what is required for Checker to run. **/
-    private static class CliOptions {
```

Comment:
Just asking, is it possible to keep this helper class and not move all fields to parent?

Reply:
Yes that’s possible. The only consideration is that by implementing `Callable` we can [reduce some boilerplate code](https://picocli.info/#_less_boilerplate), which is nice. We can make `CliOptions` implement `Callable`, and move some logic from Main to CliOptions as needed.

Reply:
@rnveach Question: if we pull all `@Option` fields into the inner class `CliOptions`, would you be okay with moving the `runCli` and `validateCli` methods to the `CliOptions` class as well? 

I like the fact that those methods can refer to the fields directly. For example:

```
if (filesToProcess.isEmpty()) // can reference fields directly
```

If  the `@Option` fields are not in the same class as the `runCli` and `validateCli` methods, we would need to pass a `CliOptions` parameter to these methods, so the code would look like this:

```
if (options.filesToProcess.isEmpty()) // referencing fields in a different class
```

This becomes a bit repetitive and I would prefer to avoid it. What do you think?